core/src/main/java/com/microsoft/applicationinsights/internal/config/TelemetryConfigurationFactory.java

```diff
@@ -327,27 +339,94 @@ private String getConfigurationFileName() {
+            return getConfFromClassPath(classLoader);
         } else {
             String configurationFile = resource.getFile();
-            InternalLogger.INSTANCE.trace("Found configuration file: '%s'", configurationFile);
+            InternalLogger.INSTANCE.logConfig(InternalLogger.LoggingLevel.INFO, "Found configuration file: '%s'", configurationFile);
             return configurationFile;
         }
+    }
+
+    private String getConfFromClassPath(ClassLoader classLoader) {
+        if (!(classLoader instanceof URLClassLoader)) {
+            return null;
+        }
+
+        HashSet<String> checkedUrls = new HashSet<String>();
+
+        URL[] urls = ((URLClassLoader)classLoader).getURLs();
+        for (URL url : urls) {
+            String configurationPath = url.getPath();
+            if (configurationPath.endsWith(".jar")) {
+                int index = configurationPath.lastIndexOf('/');
+                if (index != -1) {
+                    configurationPath = configurationPath.substring(0, index + 1);
+                } else {
+                    continue;
+                }
+            } else {
+                if (!configurationPath.endsWith("/")) {
+                    configurationPath += '/';
+                }
+            }
+
+            if (checkedUrls.contains(configurationPath)) {
+                continue;
+            }
+
+            String configurationFile = configurationFileExists(configurationPath);
+            if (configurationFile != null) {
+                return configurationFile;
+            } else {
+                checkedUrls.add(configurationPath);
+            }
+        }
+
+        return null;
+    }
+
+    private String getConfFromSourceLocation() {
+        try {
+            String jarFullPath = TelemetryConfigurationFactory.class.getProtectionDomain().getCodeSource().getLocation().toURI().getPath();
+            File jarFile = new File(jarFullPath);
+
+            if (jarFile.exists()) {
+                String jarDirectory = jarFile.getParent();
+
+                String configurationFile = configurationFileExists(jarDirectory);
+
+                if (configurationFile != null) {
+                    return jarDirectory;
+                }
+            } else {
+                InternalLogger.INSTANCE.logConfig(InternalLogger.LoggingLevel.ERROR, "Can not access folder '%s'", jarFullPath);
+            }
+        } catch (URISyntaxException e) {
+            InternalLogger.INSTANCE.logConfig(InternalLogger.LoggingLevel.ERROR, "Failed to find configuration file, exception: '%s'", e.getMessage());
+        }
+
+        return null;
+    }
+
+    private String configurationFileExists(String path) {
+        File configFile = new File(path, fileToParse);
+
+        if (configFile.exists()) {
+            InternalLogger.INSTANCE.logConfig(InternalLogger.LoggingLevel.INFO, "Found configuration file in: '%s'", path);
+            return configFile.getAbsolutePath();
+        } else {
```

Comment:
As you're returning in the `if` part there's no need for an else - just place your code after the `if` scope